lib/docassemble/docassemble_api_interface.js

```diff
@@ -0,0 +1,377 @@
+  let { wait_seconds_per_loop, max_loops } = options || { wait_seconds_per_loop: 5, max_loops: 15 };
+
+  // Flags for stopping the `while` loop
+  let num_tries = 0;
+  let done_restarting = false;
+
+  // Why loop instead of one long timeout? Because it might not just hang.
+  // We might get a return value that's just `waiting`, meaning the
+  // task is not done.
+  // - [ ] How many tries/time do we give this? Do we allow customization of those values?
+  while ( !done_restarting && num_tries <= max_loops ) {
+    num_tries++;
+
+    done_restarting = await da_i.has_task_finished( task_id );
+    if ( !done_restarting ) {
+      if ( num_tries < max_loops ) {
+        log.info({ pre: `Attempt ${ num_tries } of ${ max_loops } did not succeed. `
+          + `Will try again in ${ wait_seconds_per_loop } seconds.`
+        });
+        await time.waitForTimeout( wait_seconds_per_loop * 1000 );
+
+      } else {
+        // This is our last round, let's wrap it up
+        let server_msg = `ALKiln setup: It seems like the docassmble server `
+          + `didn't finish restarting after ${ max_loops * wait_seconds_per_loop } `
+          + `seconds. Some things you can try: check your server, try `
+          + `re-running the test, and check pulling the interview `
+          + `manually. See docs at https://suffolklitlab.org/docassemble-AssemblyLine-documentation/docs/automated_integrated_testing/.`;
+        throw( server_msg );
+      }  // ends if ran out of tries
+    }  // ends if done_restarting
+  }  // ends while not done restarting
+
+  log.info({ pre: `Package has been pulled and the docassmble server has finished restarting.` });
+};  // Ends da_i.wait_for_server_to_restart()
+
+
+// Delete the Project
+// node -e 'require("./lib/github_integration/docassemble_api_interface.js").delete_project( "create1" )'
+da_i.delete_project = async function ( delete_options, server_wait_options ) {
+  /** Delete a project on the user's da server. */
+
+  // Defaults
+  let { timeout } = delete_options ||  { timeout: 75 * 1000 };
+
+  let project_name = session_vars.get_project_name()
+  log.info({ pre: `Trying to delete Project ${ project_name }` });
+
+  try {
+
+    let response = await _da.delete_project( timeout );
+    log.info({ pre: `Deleted Project "${ project_name }"` });
+    return response.data;
+
+  } catch (error) {
+    // Should we throw in here?
+    log.debug({ pre: `See https://docassemble.org/docs/api.html#playground_delete_project.` });
+    log.info({ pre: `Ran into an error when deleting Project ${ project_name }` });
+    throw error;
+  }
+
+};  // Ends da_i.delete_project()
+
+
+// node -e 'require("./lib/github_integration/docassemble_api_interface.js").__delete_projects_starting_with( "testingdocassembleALAutomatedTestingTestsmain", 1, 1 )'
+da_i.__delete_projects_starting_with = async function ( base_name, starting_num=1, ending_num=50 ) {
+  /** Useful for development with accidental infinite loops. */
```

Comment:
We have decided to change this to "***FOR DEVELOPMENT ONLY*** Useful for manually removing a set of projects that start with a shared prefix.  This happens when an infinite loop is created."